packages/marketplace/src/components/ui/webhook-edit-modal.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useEffect } from 'react'
+import { connect } from 'react-redux'
+import {
+  Modal,
+  Button,
+  DropdownSelect,
+  Input,
+  Formik,
+  Form,
+  Content,
+  Checkbox,
+  SelectOption,
+  ModalFooter,
+  ModalBody,
+  ModalHeader,
+  Loader,
+} from '@reapit/elements'
+import { ReduxState } from '@/types/core'
+import {
+  requestWebhookSubcriptionData,
+  CreateWebhookParams,
+  EditWebhookParams,
+  createWebhook,
+  editWebhook,
+  requestWebhookData,
+} from '@/actions/webhook-edit-modal'
+import { WebhookEditState } from '@/reducers/webhook-edit-modal'
+
+const CREATE_MODAL = {
+  title: 'Add New Webhook',
+  submit: 'Create',
+}
+
+const EDIT_MODAL = {
+  title: 'Edit',
+  submit: 'Update',
+}
+
+export type WebhookEditProps = {
+  isUpdate?: boolean
+  appId: string
+  webhookId?: string
+  visible: boolean
+  closeModal?: () => void
+}
+export interface WebhookModalInnerMappedAction {
+  requestWebhookSubcriptionData: (appId: string) => void
+  requestWebhookData: (webhookId: string) => void
+  createWebhook: (data: CreateWebhookParams) => void
+  editWebhook: (data: EditWebhookParams) => void
+}
+
+export const WebhookCreateModal: React.FunctionComponent<WebhookEditProps> = ({
+  isUpdate = false,
+  appId,
+  visible,
+  webhookId,
+  closeModal,
+}) => {
+  return (
+    <Modal visible={visible} renderChildren>
+      <WebhookModalInnerWithConnect isUpdate={isUpdate} closeModal={closeModal} appId={appId} webhookId={webhookId} />
+    </Modal>
+  )
+}
+
+export const generateTopicOptions = topics => {
```

Comment:
Can we add a definition to 'topics' var - currently is of type 'any' also, looks like needs a test